Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -3586,6 +3586,11 @@ void WebPage::mouseEvent(FrameIdentifier frameID, const WebMouseEvent& mouseEven
     }
 
     send(Messages::WebPageProxy::DidReceiveEvent(mouseEvent.type(), handled, std::nullopt));
+
+#if PLATFORM(IOS_FAMILY)
+    if (m_page->settings().visuallyContiguousBidiTextSelectionEnabled() && mouseEvent.type() == WebEventType::MouseUp)
+        removeTextInteractionSources(TextInteractionSource::Mouse);
```

Comment:
Is this intentionally gated on the `visuallyContiguousBidiTextSelectionEnabled ` setting? The call to `addTextInteractionSources` isn't, so this probably shouldn't be either?

Reply:
Oh, good catch — I forgot to remove the settings-check here (I'd originally made this option set specific to this feature, but then changed my mind when I realized it could be useful for other things in the future)